### DISCUSSION TITLE
404 error on a href link?

### DISCUSSION THREAD

**Discussion by vishnukmurthy** [community/community#21645] [Other Feature Feedback, Questions, & Ideas]
<p>I don’t know what’s happening, but when I run my site locally everything looks ok, however when I push to github pages I keep getting a 404. I am using relative paths and am confused because I’m doing the same thing for multiple pages.</p>
<p>I.e. I have a /booknotes folder that works fine.<br>
and I have a /blog folder where everything seems to break.</p>
<p>It’s like the relative link doesn’t work just for that folder. But It’s all fine locally.</p>
<p>Here is where my code is hosted: <a href="https://github.com/vishnukmurthy/vishnukmurthy.github.io" class="inline-onebox" rel="noopener nofollow ugc">GitHub - vishnukmurthy/vishnukmurthy.github.io</a></p>

**Comment by jfullstackdev** [comment]
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/9/3/9335ecab7071875b2ac293da64a43d18f8c0af26.jpeg" data-download-href="/uploads/short-url/l0hIGVyFZTEkS4Yy73jKBGwmsCO.jpeg?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/47092464/181065974-42f9d7e0-a1d0-4c6f-92f5-e3c3dbf696af.jpeg" alt="image" data-base62-sha1="l0hIGVyFZTEkS4Yy73jKBGwmsCO" width="690" height="387" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/9/3/9335ecab7071875b2ac293da64a43d18f8c0af26_2_690x387.jpeg, https://aws1.discourse-cdn.com/github/optimized/3X/9/3/9335ecab7071875b2ac293da64a43d18f8c0af26_2_1035x580.jpeg 1.5x, https://aws1.discourse-cdn.com/github/original/3X/9/3/9335ecab7071875b2ac293da64a43d18f8c0af26.jpeg 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/9/3/9335ecab7071875b2ac293da64a43d18f8c0af26_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1366×768 172 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>it’s working now, have you checked it? I just checked it now,</p>
<p>when you update or build something, it will take time, maybe 5-10 minutes</p>

**Comment by vishnukmurthy** [comment]
<p>hmm, no it’s still broken. Maybe I should be more clear about my issue.<br>
If I click the “blog” tab, , and then click any of the blog posts you will see a 404.<br>
However, if you click the “book notes” tab, and then click on a book note everything works.</p>
<p>The code is structured similarly. I.e. blog is a folder and book notes is a folder, and the posts are within that folder. References to both are done as absolute paths.</p>

**Comment by jfullstackdev** [comment, marked as answer]
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/3/f/3fb1d18fba2bc7dfdd5e3a75b100813494a0671a.png" data-download-href="/uploads/short-url/95t3qiWuzuFyZurikfgpmDM2S4W.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/47092464/181065975-92526f47-a528-471f-9b63-dc79ae6db16a.png" alt="image" data-base62-sha1="95t3qiWuzuFyZurikfgpmDM2S4W" width="690" height="387" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/3/f/3fb1d18fba2bc7dfdd5e3a75b100813494a0671a_2_690x387.png, https://aws1.discourse-cdn.com/github/optimized/3X/3/f/3fb1d18fba2bc7dfdd5e3a75b100813494a0671a_2_1035x580.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/3/f/3fb1d18fba2bc7dfdd5e3a75b100813494a0671a.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/3/f/3fb1d18fba2bc7dfdd5e3a75b100813494a0671a_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1366×768 83.6 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>the crime and punishment in <code>booknotes</code> folder is working, meaning, your <code>blog.html</code> at the <code>main</code> folder is causing the issue, clashing with <code>blog</code> folder. You can simply put an <code>s</code> so that it will be a different word.</p>

**Comment by vishnukmurthy** [comment]
<p>First off, thanks for the help! I really appreciate it. This fixed my issue!</p>
<p>I do have a few follow up questions that I hope you wouldn’t mind taking the time to answer though.</p>
<ol>
<li>
<p>I had a booknotes.html and a /booknotes folder. Why did the issue only happen for blog.html and /blog and not for the booknotes equivalent? The logic seems the same so I imagine the issue should persist? The only difference is booknotes has only one file in the directory while the blog folder has multiple.</p>
</li>
<li>
<p>Is this a bug? I see these tickets that have been ‘frozen due to age’<br>
<a href="https://github.com/jekyll/jekyll/issues/6475" class="inline-onebox" rel="noopener nofollow ugc">If `bar.html` exists, then `jekyll s` will show `bar.html` for `bar/.*` unless something else explicitly (but not implicitly) overrides it · Issue #6475 · jekyll/jekyll · GitHub</a><br>
<a href="https://github.com/jekyll/jekyll/issues/6459" class="inline-onebox" rel="noopener nofollow ugc">`jekyll s` redirects to slightly wrong URL (with file and folder of same name) · Issue #6459 · jekyll/jekyll · GitHub</a><br>
My ruby game is weak, but I think it would make sense for me to try and help fix the issue if possible haha.</p>
</li>
</ol>
<p>Thanks again for all your help.</p>

**Comment by jfullstackdev** [comment]
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/2/0/20984fa5a9d8dc075c029e8746158ed91032d6dd.jpeg" data-download-href="/uploads/short-url/4ElBsgsEX7QS0U5scVkecTuD5cp.jpeg?dl=1" title="Screen Shot 2021-04-26 at 8.26.32 AM" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/47092464/181065978-35e9cbed-eeac-4953-b739-08760475daf2.jpeg" alt="Screen Shot 2021-04-26 at 8.26.32 AM" data-base62-sha1="4ElBsgsEX7QS0U5scVkecTuD5cp" width="587" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/2/0/20984fa5a9d8dc075c029e8746158ed91032d6dd_2_587x500.jpeg, https://aws1.discourse-cdn.com/github/optimized/3X/2/0/20984fa5a9d8dc075c029e8746158ed91032d6dd_2_880x750.jpeg 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/2/0/20984fa5a9d8dc075c029e8746158ed91032d6dd_2_1174x1000.jpeg 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/2/0/20984fa5a9d8dc075c029e8746158ed91032d6dd_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screen Shot 2021-04-26 at 8.26.32 AM</span><span class="informations">2680×2280 402 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>this is tricky, you are using some kind of template</p>
<p><a href="https://vishnukmurthy.com/blog/" rel="noopener nofollow ugc">https://vishnukmurthy.com/blog/</a></p>
<p>that’s the page when it’s just <code>blog</code> but when you have the complete,</p>
<p><a href="https://vishnukmurthy.com/blog.html" rel="noopener nofollow ugc">https://vishnukmurthy.com/blog.html</a></p>
<p>this is your site</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/8/7/87ec3e1cf9eb86ebd1d95faa958ab8d0a72c7d5e.jpeg" data-download-href="/uploads/short-url/joqBzvdRYH144REJNZ5jXr4Xsjc.jpeg?dl=1" title="Screen Shot 2021-04-26 at 8.30.21 AM" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/47092464/181065979-c39c5d45-d52b-423d-bd86-d419d9ec25f6.jpeg" alt="Screen Shot 2021-04-26 at 8.30.21 AM" data-base62-sha1="joqBzvdRYH144REJNZ5jXr4Xsjc" width="587" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/8/7/87ec3e1cf9eb86ebd1d95faa958ab8d0a72c7d5e_2_587x500.jpeg, https://aws1.discourse-cdn.com/github/optimized/3X/8/7/87ec3e1cf9eb86ebd1d95faa958ab8d0a72c7d5e_2_880x750.jpeg 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/8/7/87ec3e1cf9eb86ebd1d95faa958ab8d0a72c7d5e_2_1174x1000.jpeg 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/8/7/87ec3e1cf9eb86ebd1d95faa958ab8d0a72c7d5e_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screen Shot 2021-04-26 at 8.30.21 AM</span><span class="informations">2680×2280 549 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>and the last one to take note of</p>
<p><a href="https://vishnukmurthy.com/blog.html/" rel="noopener nofollow ugc">https://vishnukmurthy.com/blog.html/</a></p>
<p>when you put <code>/</code>, it is 404</p>

**Comment by vishnukmurthy** [comment]
<p>Ah. I think I know what the problem is. Github Pages - where my website is being hosted runs Jekyll on the backend behind the scenes. It looks like /blog/ has special meaning in Jekyll and things get funky if I name a folder that.</p>

**Comment by jfullstackdev** [comment]
<p>yup, but I’m not so sure of all the details of your page, I’m just observing it from the external</p>

**Comment by vishnukmurthy** [comment]
<p>appreciate the help. Anyway that I can mark your answer as a solution, or give you official credit via github? This is my first time using this board.</p>

**Comment by jfullstackdev** [comment]
<p>yup, no problem with that, thanks, just mark it as a solution</p>

**Comment by vishnukmurthy** [comment]
<p>any chance you could tell me how to do that? I don’t see any intuitive way, all I see is like and flag as options :confused:</p>

**Comment by jfullstackdev** [comment]
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/5/9/597689b141af92a525c0421e17c07629b187f5c8.png" data-download-href="/uploads/short-url/cLqsAkTpWGmBvZ7CekcQw7eF1t6.png?dl=1" title="Screen Shot 2021-04-27 at 1.33.07 PM" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/47092464/181065984-e3cee581-2551-4ffb-a0f1-164682d5e126.png" alt="Screen Shot 2021-04-27 at 1.33.07 PM" data-base62-sha1="cLqsAkTpWGmBvZ7CekcQw7eF1t6" width="690" height="495" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/5/9/597689b141af92a525c0421e17c07629b187f5c8_2_690x495.png, https://aws1.discourse-cdn.com/github/optimized/3X/5/9/597689b141af92a525c0421e17c07629b187f5c8_2_1035x742.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/5/9/597689b141af92a525c0421e17c07629b187f5c8_2_1380x990.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/5/9/597689b141af92a525c0421e17c07629b187f5c8_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screen Shot 2021-04-27 at 1.33.07 PM</span><span class="informations">1638×1176 213 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>if you place your cursor to the <code>Solution</code> then click it, that’s it, at the bottom, near the reaction button</p>

**Comment by otter-computer** [comment]
<p>Hi <a class="mention" href="/vishnukmurthy">@vishnukmurthy</a>! It looks like you couldn’t mark this as solved because it wasn’t inside a category. I’ve moved this thread to the GitHub Pages category now and marked <a class="mention" href="/jdevstatic">@jdevstatic</a>’s response as the solution for you :sparkles:</p>

**Comment by jfullstackdev** [comment]
<p>thanks <a class="mention" href="/thomasshaped">@thomasshaped</a> and <a class="mention" href="/vishnukmurthy">@vishnukmurthy</a></p>
